### PR TITLE
dumpling: compress supports snappy/zstd

### PR DESCRIPTION
**Pull request by lyzx2001**
### What problem does this PR solve?

Issue Number: ref #38514 

Problem Summary:

### What is changed and how it works?
- [x] Dumpling compress supports snappy/zstd

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by lyzx2001** [comment]
> Is this PR ready for review now?

there are still 2 tests failing and I am trying to fix them

File: dumpling/export/config.go [hunk 1]
```diff
@@ -593,6 +593,10 @@ func ParseCompressType(compressType string) (storage.CompressType, error) {
 		return storage.NoCompression, nil
 	case "gzip", "gz":
 		return storage.Gzip, nil
+	case "snappy":
+		return storage.Snappy, nil
+	case "zst":
```
Comment (lichunzhu) [inline dumpling/export/config.go:598]: Do you mean this? It's different with the description in comment
```suggestion
	case "zstd":
```
or maybe
```suggestion
	case "zstd","zst":
```

File: dumpling/tests/s3/run.sh [hunk 2]
```diff
@@ -97,4 +98,44 @@ diff "${HOST_DIR}/local/s3.t-schema.sql" "${HOST_DIR}/compress/s3.t-schema.sql"
...
+bin/mc cp minio/mybucket/dump-compress/s3.t.000000000.sql.zst "${HOST_DIR}/compress/s3.t.000000000.sql.zst"
+
+zstd "${HOST_DIR}/compress/s3-schema-create.sql.zst" -d
+diff "${HOST_DIR}/local/s3-schema-create.sql" "${HOST_DIR}/compress/s3-schema-create.sql"
+
+zstd "${HOST_DIR}/compress/s3.t-schema.sql.zst" -d
+diff "${HOST_DIR}/local/s3.t-schema.sql" "${HOST_DIR}/compress/s3.t-schema.sql"
+
+zstd "${HOST_DIR}/compress/s3.t.000000000.sql.zst" -d
+diff "${HOST_DIR}/local/s3.t.000000000.sql" "${HOST_DIR}/compress/s3.t.000000000.sql"
```
Comment (lichunzhu) [inline dumpling/tests/s3/run.sh:139]: Can we merge these three parts in a for loop?

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM

**Review by lichunzhu — APPROVED** [review]
Great job! LGTM

**Comment by lance6716** [comment]
will review later

File: dumpling/install.sh [hunk 3]
```diff
@@ -15,3 +15,11 @@ chmod a+x bin/minio
 
 wget https://dl.minio.io/client/mc/release/linux-amd64/mc -O bin/mc
 chmod a+x bin/mc
+
+go get github.com/ma6174/snappy
```
Comment (lance6716) [inline dumpling/install.sh:19]:
```suggestion
```

only need go install?

File: dumpling/tests/s3/run.sh [hunk 4]
```diff
@@ -80,21 +80,58 @@ echo "1st records count is ${cnt}"
...
+  else
+    compressFormat=$compressType
+  fi
+  bin/mc cp minio/mybucket/dump-compress/s3-schema-create.sql.${compressFormat} "${HOST_DIR}/compress/s3-schema-create.sql.${compressFormat}"
+  bin/mc cp minio/mybucket/dump-compress/s3.t-schema.sql.${compressFormat} "${HOST_DIR}/compress/s3.t-schema.sql.${compressFormat}"
+  bin/mc cp minio/mybucket/dump-compress/s3.t.000000000.sql.${compressFormat} "${HOST_DIR}/compress/s3.t.000000000.sql.${compressFormat}"
+
+  case $compressType in
+  "gzip")
+    gzip "${HOST_DIR}/compress/s3-schema-create.sql.gz" -d
```
Comment (lance6716) [inline dumpling/tests/s3/run.sh:101]: seems we can assign gzip, snappy, zst to `$binary`, and also assign $suffix and use

```bash
    ${binary} "${HOST_DIR}/compress/s3-schema-create.sql.${suffix}" -d
    diff "${HOST_DIR}/local/s3-schema-create.sql" "${HOST_DIR}/compress/s3-schema-create.sql"

    ${binary} "${HOST_DIR}/compress/s3.t-schema.sql.${suffix}" -d
    diff "${HOST_DIR}/local/s3.t-schema.sql" "${HOST_DIR}/compress/s3.t-schema.sql"

    ${binary} "${HOST_DIR}/compress/s3.t.000000000.sql.${suffix}" -d
    diff "${HOST_DIR}/local/s3.t.000000000.sql" "${HOST_DIR}/compress/s3.t.000000000.sql"
```

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: dumpling/tests/s3/run.sh (hunk 4 above)
Comment (lyzx2001) [reply dumpling/tests/s3/run.sh]: modified

**Review by lance6716 — APPROVED** [review]

File: dumpling/install.sh (hunk 3 above)
Comment (lyzx2001) [reply dumpling/install.sh]: error encountered in CI integration test

https://github.com/pingcap/tidb/actions/runs/3444134903/jobs/5746413416#step:6:7138

i think maybe should back `go get github.com/ma6174/snappy`?

Comment (lyzx2001) [reply dumpling/install.sh]: error encountered in CI integration test

https://github.com/pingcap/tidb/actions/runs/3444134903/jobs/5746413416#step:6:7138

i think maybe should add back `go get github.com/ma6174/snappy`?

**Pull request merged**